parser/src/lexer.rs

```diff
@@ -644,6 +696,12 @@ where
                     spaces = 0;
                     tabs = 0;
                 }
+                // https://github.com/ipython/ipython/blob/635815e8f1ded5b764d66cacc80bbe25e9e2587f/IPython/core/inputtransformer2.py#L345
+                Some('%' | '!' | '?' | '/' | ';' | ',') if self.mode == Mode::Jupyter => {
+                    self.lex_and_emit_magic_command()?;
+                    spaces = 0;
+                    tabs = 0;
```

Comment:
I don't think these are necessary here. The magic command doesn't eat the trailing new line. Meaning, `spaces` and `tabs` are set to `zero` when processing the `newline`.

Reply:
Right, does it mean it isn't required for comments as they are consumed till end of line as well?

Reply:
The main reason why I think it isn't required is because the magic command lexing eats everything up to the new line. The next `next` call then lexes a newline, which resets the spaces and tabs. But I might be wrong.

Reply:
Yes, I'm thinking the same, will test it out once and then confirm.

Reply:
This is correct. We don't need to reset the indentation (it's true for comments as well but it's out of scope for this PR).